src/odemis/driver/actuator.py

```diff
@@ -2806,3 +2806,332 @@ def _doReference(self, future, axes):
+
+        # position will be calculated from the underlying stage
+        self.position = model.VigilantAttribute({}, readonly=True)
+        self._stage.position.subscribe(self._updatePosition, init=True)
+
+        # Add speed if it's in stage
+        if model.hasVA(self._stage, "speed"):
+            self.speed = self._stage.speed
+
+    def _validateDepStage(self, dependencies):
+        """
+        Make sure there is only one actuator dependency with the right axes
+        """
+        if len(dependencies) != 1:
+            raise ValueError("LinkedAxesActuator needs precisely one dependency")
+        crole, dep = dict(dependencies).popitem()
+        # Check if dependency is indeed actuator
+        if not isinstance(dep, model.ComponentBase):
+            raise ValueError("Dependency %s is not a component." % dep)
+        if not hasattr(dep, "axes") or not isinstance(dep.axes, dict):
+            raise ValueError("Dependency %s is not an actuator." % dep.name)
+        # Check if dependency has the right axes
+        elif crole == "stage":
+            if not {'x', 'y', 'z'}.issubset(dep.axes.keys()):
+                raise ValueError("Dependency %s doesn't have x, y and z axes" % dep.name)
+            self._stage = dep
+        else:
+            raise ValueError(
+                "Dependency given to LinkedAxesActuator must be 'stage', but got %s." % crole)
+
+    def _updatePosition(self, value=None):
+        """
+        update the position VA by mapping from the underlying stage
+        """
+        # Only update position when MD_POS_COR and MD_CALIB are already configured
+        if not {model.MD_POS_COR, model.MD_CALIB}.issubset(self._metadata.keys()):
+            logging.error("POS_COR and CALIB should be in metadata.")
+            return
+
+        value = self._stage.position.value if value is None else value
+        # Map x, y from the underlying stage
+        x, y = self._computeLinkedAxes(value)
+        current_value = {'x': x, 'y': y}
+        logging.debug("Updating linked axes position %s", current_value)
+        self.position._set_value(current_value, force_write=True)
+
+    def updateMetadata(self, md):
+        """
+        Validate and update the metadata associated with the axes mapping
+        md (dict string -> value): the metadata
+        """
+        self._validateMetadata(md)
+        super(LinkedAxesActuator, self).updateMetadata(md)
+        # Re-update position whenever MD_POS_COR and MD_CALIB values change
+        if len({model.MD_POS_COR, model.MD_CALIB}.intersection(md.keys())):
+            self._updatePosition()
+
+    def _validateMetadata(self, md):
+        """
+        Ensure MD_POS_COR and MD_CALIB have the right parameters for axes mapping
+        :raises ValueError: if parameters are incorrect
+        """
+        if model.MD_POS_COR in md and len(md[model.MD_POS_COR]) != 3:
+            raise ValueError("MD_POS_COR should have 3 parameters.")
```

Comment:
Note: In "_metadata.py" MD_POS_COR is commented as a tuple of length 2:
> MD_POS_COR = "Centre position cor"  # (m, m), to be subtracted from MD_POS

Just FYI I don't know if this matters or should be changed, neither if an extra comment is needed in your code to clearify this.